lib/fluent/plugin/s3_compressor_zstd.rb

```diff
@@ -0,0 +1,36 @@
+require 'zstd-ruby'
+
+module Fluent::Plugin
+  class S3Output
+    class ZstdCompressor < Compressor
+      S3Output.register_compressor('zstd', self)
+
+      config_section :compress, param_name: :compress_config, init: true, multi: false do
+        desc "Compression level for zstd (1-22)"
+        config_param :level, :integer, default: 3
+      end
+
+      def ext
+        'zst'.freeze
+      end
+
+      def content_type
+        'application/x-zst'.freeze
+      end
+
+      def compress(chunk, tmp)
+        w = StringIO.new
+        chunk.write_to(w)
+        w.rewind
+        compressed = Zstd.compress(w.read, level: @compress_config.level)
+        tmp.binmode
+        tmp.rewind
+        tmp.write(compressed)
+        tmp.rewind
```

Comment:
Thanks for the fix!
Now, looks good to me except for the `compress` method.

> The real issue was not with line endings or encoding, but with the compression implementation itself. After fixing the compression logic to use `chunk.write_to` method properly, the compression and tests are now working correctly without any need for explicit line ending or encoding manipulation.

On my fork, all `S3OutputTest` tests pass with the following simple implementation.

```suggestion
      def compress(chunk, tmp)
        compressed = Zstd.compress(chunk.read, level: @compress_config.level)
        tmp.write(compressed)
```

I would like to know which cases cannot be handled correctly with this simple implementation.
Do you use Windows?
When I run the test on Windows, the line ending code is different from the expected value.

Reply:
@daipom 
Thank you for the suggestion! You're right - your simpler implementation works perfectly.

I initially overcomplicated the implementation with unnecessary buffer handling and line ending concerns. Your version is much cleaner and achieves the same result. I tested your simplified version and all tests pass successfully.

I don't use Windows, and my concerns about line endings were unfounded. I was trying to solve compression issues by handling line endings when the real solution was much simpler.

I'll update the code to use your suggested implementation. It's more straightforward and easier to understand.

Reply:
Thanks!